opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1678,8 +1678,17 @@ namespace Opm
             const auto report = getWellConvergence(simulator, well_state, Base::B_avg_, deferred_logger, relax_convergence);
             converged = report.converged();
             if (this->parallel_well_info_.communication().size() > 1 && converged != this->parallel_well_info_.communication().min(converged)) {
-                OPM_THROW(std::runtime_error, fmt::format("Misalignment of the parallel simulation run in iterateWellEqWithSwitching - the well calculation succeeded on rank {} but failed on other ranks.", this->parallel_well_info_.communication().rank()));
+                std::cout << fmt::format("Misalignment of the parallel simulation run in iterateWellEqWithSwitching - the well calculation for well {} succeeded on rank {} but failed on other ranks.", this->name(), this->parallel_well_info_.communication().rank()) << std::endl;
+                OpmLog::warning(fmt::format("Misalignment of the parallel simulation run in iterateWellEqWithSwitching - the well calculation  for well {} succeeded on rank {} but failed on other ranks.", this->name(), this->parallel_well_info_.communication().rank()));
             }
+            if (this->parallel_well_info_.communication().size() > 1 && converged != this->parallel_well_info_.communication().max(converged)) {
+                std::cout << fmt::format("Misalignment of the parallel simulation run in iterateWellEqWithSwitching - the well calculation  for well {} failed on rank {} but succeeded on other ranks.", this->name(), this->parallel_well_info_.communication().rank()) << std::endl;
+                OpmLog::warning(fmt::format("Misalignment of the parallel simulation run in iterateWellEqWithSwitching - the well calculation  for well {} failed on rank {} but succeeded on other ranks.", this->name(), this->parallel_well_info_.communication().rank()));
+            }
+            if (this->parallel_well_info_.communication().size() > 1 && (converged != this->parallel_well_info_.communication().min(converged) || converged != this->parallel_well_info_.communication().max(converged))) {
```

Comment:
I'm not sure that this will work. The second comparison might yield different results on different processes. In that case the latest communication will not be triggered on all processes.

So maybe move the communication outside of the logical expression and compare `this->parallel_well_info_.communication().min(converged)` with `this->parallel_well_info_.communication().max(converged)`?